python/lsst/ip/isr/calibType.py

```diff
@@ -205,6 +218,63 @@ def updateMetadata(self, camera=None, detector=None, filterName=None,
         mdSupplemental.update(kwargs)
         mdOriginal.update(mdSupplemental)
 
+    def calibInfoFromDict(self, dictionary):
+        """Handle common keywords.
+
+        This isn't an ideal solution, but until all calibrations
+        expect to find everything in the metadata, they still need to
+        search through dictionaries.
+
+        Parameters
+        ----------
+        dictionary : `dict` or `lsst.daf.base.PropertyList`
+            Source for the common keywords.
+
+        Raises
+        ------
+        RuntimeError :
+            Raised if the dictionary does not match the expected OBSTYPE.
+
+        """
+
+        def search(haystack, needles):
+            """Search dictionary 'haystack' for an entry in 'needles'
+            """
+            test = [haystack.get(x) for x in needles]
+            test = set([x for x in test if x is not None])
+            if len(test) == 0:
+                if 'metadata' in haystack:
+                    return search(haystack['metadata'], needles)
+                else:
+                    return None
+            elif len(test) == 1:
+                value = list(test)[0]
+                if value == '':
+                    return None
+                else:
+                    return value
+            else:
+                raise ValueError(f"Too many values found: {len(test)} {test} {needles}")
```

Comment:
I would likely have had comments here, but we agreed this OOB before hand, so that is fine for now.

Reply:
I am open to other more elegant solutions, but I don't know that that can exist until some future point where all of the metadata keywords are standardized completely.